src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -216,9 +226,20 @@ export class ProsemirrorAdapter {
         this.view.dispatch(tr);
     }
 
-    private handleTransaction = (transaction) => {
+    private handleTransaction = (transaction: Transaction) => {
         const newState = this.view.state.apply(transaction);
         this.view.updateState(newState);
+
+        if (transaction.getMeta('pointer')) {
+            return;
+        }
+
+        if (isEqual(this.prosemirrorDoc, transaction.doc) && this.shouldEmit) {
```

Comment:
Logical operators are evaluated left-to-right. Since `this.shouldEmit` will _almost always_ be `false`, we can save a lot of work by not recursively comparing the two data structures of unknown size just to then throw the result away when `this.shouldEmit` is evaluated to false.

```suggestion
        if (this.shouldEmit && isEqual(this.prosemirrorDoc, transaction.doc)) {
```

By putting the simple boolean check first, we will never enter the deep equality check if the boolean is `false`.